dll/win32/shell32/shlfileop.cpp

```diff
@@ -795,6 +810,14 @@ static DWORD SHNotifyCopyFileW(FILE_OPERATION *op, LPCWSTR src, LPCWSTR dest, BO
     ret = CopyFileExW(src, dest, SHCopyProgressRoutine, op, &op->bCancelled, bFailIfExists);
     if (ret)
     {
+        // We are copying from a CD-ROM volume, which is readonly
+        if (SHIsCdRom(src))
```

Comment:
If a ton of files are moved from the same origin (CD-ROM), is this function going to be called such a ton of times?

Reply:
yes, but we cannot just call this on the first file, because it's possible to copy files from multiple locations

Reply:
ok if the shell doesn't group the file operations also by source folder/device, then indeed I don't see how we can improve that.